rasa/cli/data.py

```diff
@@ -109,4 +150,51 @@ def validate_files(args) -> NoReturn:
         sys.exit(1)
 
     everything_is_alright = validator.verify_all(not args.fail_on_warnings)
-    sys.exit(0) if everything_is_alright else sys.exit(1)
+    if not args.max_history:
+        logger.info(
+            "Will not test for inconsistencies in stories since "
+            "you did not provide a value for `--max-history`."
+        )
+    else:
+        # Only run story structure validation if everything else is fine
+        # since this might take a while
+        everything_is_alright = validator.verify_story_structure(
+            not args.fail_on_warnings, max_history=args.max_history
+        )
+
+    if not everything_is_alright:
+        print_error("Project validation completed with errors.")
+        sys.exit(1)
+
+
+def validate_stories(args: "argparse.Namespace") -> None:
```

Comment:
this is duplicate to `validate_files`, isn't it?

Reply:
No, `validate_files` validates everything, whereas `validate_stories` only validates the story structure. I've now updated the doc string to clarify.